src/main/groovy/io/seqera/wave/service/blob/impl/BlobCacheServiceImpl.groovy

```diff
@@ -63,19 +69,25 @@ class BlobCacheServiceImpl implements BlobCacheService {
 
     private HttpClient httpClient
 
+    private S3Presigner presigner
+
     @PostConstruct
     private void init() {
         httpClient = HttpClientFactory.followRedirectsHttpClient()
+        presigner = S3Presigner.builder()
+                .region(  Region.of(blobConfig.storageRegion.toUpperCase()) )
+                .credentialsProvider(DefaultCredentialsProvider.create())
+                .build()
         log.info "Creating Blob cache service - $blobConfig"
     }
 
     @Override
     BlobCacheInfo retrieveBlobCache(RoutePath route, Map<String,List<String>> headers) {
-        final uri = blobDownloadUri(route)
-        final info = BlobCacheInfo.create(uri, headers)
+        final info = BlobCacheInfo.create(headers)
         final target = route.targetPath
         if( blobStore.storeIfAbsent(target, info) ) {
             // start download and caching job
+            info.locationUri = blobDownloadUri(route)
```

Comment:
But the problem is the `return awaitCacheStore` branch. Let's keep this 👇 as it is 

```
        final uri = blobDownloadUri(route)
        final info = BlobCacheInfo.create(uri, headers)
```

and pass the `uri` as parameter to `awaitCacheStore` to update he blob info returned by it

Reply:
done